pkg/decision/evaluator/matchers/gt.go

```diff
@@ -22,18 +22,33 @@ import (
 
 	"github.com/optimizely/go-sdk/pkg/decision/evaluator/matchers/utils"
 	"github.com/optimizely/go-sdk/pkg/entities"
+	"github.com/optimizely/go-sdk/pkg/logging"
 )
 
 // GtMatcher matches against the "gt" match type
-func GtMatcher(condition entities.Condition, user entities.UserContext) (bool, error) {
+func GtMatcher(condition entities.Condition, user entities.UserContext, logger logging.OptimizelyLogProducer) (bool, error) {
+	return matchGtOrLt(condition, user, logger, true)
+}
+
+func matchGtOrLt(condition entities.Condition, user entities.UserContext, logger logging.OptimizelyLogProducer, gtMatch bool) (bool, error) {
```

Comment:
Nit, but If we make this a pure comparator and return `(int, bool)` e.g. -1, 0, 1, then we can drop the `gtMatch bool` which is a bit awkward. Then consumers would look like:
```go
func GtMatcher(condition entities.Condition, user entities.UserContext, logger logging.OptimizelyLogProducer) (bool, error) {
	res, err := compare(condition, user, logger)
	if err := nil {
		return false, err
	}
	return res > 0, nil
}

func LtMatcher(condition entities.Condition, user entities.UserContext, logger logging.OptimizelyLogProducer) (bool, error) {
	res, err := compare(condition, user, logger)
	if err := nil {
		return false, err
	}
	return res < 0, nil
}
```
This would match the logic in the semver PR  [here](https://github.com/optimizely/go-sdk/pull/289/files#diff-10bf306c78e43b7fe2b8736f4c50d92bR163-R197)